### PR TITLE
Image manager refactor

### PR DESCRIPTION
**Pull request by astoycos**
Refactor the image manager
- Remove all channels and commands
- only initialize image_manager when needed and do this in the core codepath (bpf.rs)
- Cleanup a bunch of extra async functions which don't need to be async any longer

### CODE REVIEW

File: bpfman/src/utils.rs [hunk 1]
```diff
@@ -20,16 +21,14 @@ use crate::errors::BpfmanError;
 pub(crate) const SOCK_MODE: u32 = 0o0660;
 
 // Like tokio::fs::read, but with O_NOCTTY set
-pub(crate) async fn read<P: AsRef<Path>>(path: P) -> Result<Vec<u8>, BpfmanError> {
+pub(crate) fn read<P: AsRef<Path>>(path: P) -> Result<Vec<u8>, BpfmanError> {
     let mut data = vec![];
-    tokio::fs::OpenOptions::new()
+    std::fs::OpenOptions::new()
```
Comment (dave-tucker) [inline bpfman/src/utils.rs:34]: No need to fully qualify this. May as well `use std::fs::OpenOptions;`

File: bpfman/src/utils.rs [hunk 2]
```diff
@@ -47,22 +46,22 @@ pub(crate) fn get_ifindex(iface: &str) -> Result<u32, BpfmanError> {
     }
 }
 
-pub(crate) async fn set_file_permissions(path: &Path, mode: u32) {
+pub(crate) fn set_file_permissions(path: &Path, mode: u32) {
     // Set the permissions on the file based on input
-    if (tokio::fs::set_permissions(path, std::fs::Permissions::from_mode(mode)).await).is_err() {
+    if (std::fs::set_permissions(path, std::fs::Permissions::from_mode(mode))).is_err() {
```
Comment (dave-tucker) [inline bpfman/src/utils.rs:59]: As above re: qualification. Just import instead.

File: bpfman/src/bpf.rs [hunk 3]
```diff
@@ -54,22 +50,31 @@ const MAPS_USED_BY_PREFIX: &str = "map_used_by_";
...
         }
     }
 
+    // Make sure to call init_image_manger if the command requires interation with
+    // an OCI based container registry. It should ONLY be used where needed, to
+    // explicitly control when bpfman blocks for network calls to both sigstore's
+    // cosign tuf registries and container registries.
+    pub(crate) async fn init_image_manager(&mut self) {
+        let config = open_config_file();
+        self.image_manager = Some(
```
Comment (dave-tucker) [inline bpfman/src/bpf.rs:71]: This should only happen if `self.image_manager.is_none()`.
I know you've put comments here, but humans make mistakes.

**Review by dave-tucker — CHANGES_REQUESTED** [review]
A couple of nits but overall LGTM.

File: bpfman/src/utils.rs (hunk 2 above)
Comment (astoycos) [reply bpfman/src/utils.rs]: Done

File: bpfman/src/bpf.rs (hunk 3 above)
Comment (astoycos) [reply bpfman/src/bpf.rs]: Fair enough, done!

**Review by dave-tucker — APPROVED** [review]

**Comment by dave-tucker** [comment]
@astoycos needs a rebase!

**Pull request merged**